File: cpp/2579.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> st, dp;

int main() {
	int N;
	cin >> N;
	st.resize(N + 1, -1);
	dp.resize(N + 1, -1);
	st[0] = dp[0] = 0;
	for (int i = 1; i <= N; i++)
		cin >> st[i];
	dp[1] = st[1];
	dp[2] = st[1] + st[2];
	dp[3] = max(st[1], st[2]) + st[3];

	for (int i = 4; i <= N; i++)
		dp[i] = max(dp[i - 2], dp[i - 3] + st[i - 1]) + st[i];

	cout << dp[N];
	return 0;
}

File: python/2579.py
N = int(input())
value = [-1]*(N+1)
dp = [-1]*(N+1)

for i in range(1, N+1):
    value[i] = int(input())

def function(n):
    if dp[n]!=-1:
        return dp[n]
    dp[n] = max(function(n-2),function(n-3)+value[n-1])+value[n]
    return dp[n]

if N==1:
    print(value[1])
elif N==2:
    print(value[1]+value[2])
elif N==3:
    print(max(value[1],value[2])+value[3])
else:
    dp[1] = value[1]
    dp[2] = value[1]+value[2]
    dp[3] = max(value[1],value[2])+value[3]
    print(function(N))
          
